JavaScript/JavaScriptSDK/ajax/ajax.ts

```diff
@@ -1,288 +1,55 @@
-
-                //<summary>True, if onreadyStateChangeCallback function attached to xhr, otherwise false</summary>
-                this.onreadystatechangeCallbackAttached = false;
-
-                //<summary>Determines whether or not JavaScript exception occured in xhr.onreadystatechange code. 1 if occured, otherwise 0.</summary>
-                this.clientFailure = 0;
-
-                this.getAbsoluteUrl = function () {
-                    return this.requestUrl ? Util.getAbsoluteUrl(this.requestUrl) : null;
-                }
-
-                this.CalculateMetrics = function () {
-                    var self = this;
-                    self.ttfb = dateTime.GetDuration(self.requestSentTime, self.responseStartedTime);
-                    self.responseReceivingDuration = dateTime.GetDuration(self.responseStartedTime, self.responseFinishedTime);
-                    self.callbackDuration = dateTime.GetDuration(self.responseFinishedTime, self.callbackFinishedTime);
-
-                    var timeStamps = [self.responseStartedTime, self.responseFinishedTime, self.callbackFinishedTime];
-                    for (var i = timeStamps.length - 1; i >= 0; i--) {
-                        if (timeStamps[i] !== null) {
-                            self.endTime = timeStamps[i];
-                            self.ajaxTotalDuration = dateTime.GetDuration(self.requestSentTime, self.endTime);
-                            break;
-                        }
+            var initiated = false;
+            this.Init();
                     }
-                };
-            };
-
-            var Util = {
-                getAbsoluteUrl: (function () {
-                    var a;
-
-                    return function (url) {
-                        if (!a) a = document.createElement('a');
-                        a.href = url;
-
-                        return a.href;
-                    };
-                })()
-            }
-
-            ///<summary>The function object that provides ajax monitoring on the page</summary>
-            function ajaxMonitoring(appInsights: Microsoft.ApplicationInsights.AppInsights) {
-                var appInsights = appInsights;
-
-                var initiated = false;
 
                 ///<summary>The main function that needs to be called in order to start Ajax Monitoring</summary>
-                this.Init = function () {
-                    if (supportMonitoring()) {
-                        interceptOpen();
-                        interceptSetRequestHeader();
-                        interceptSend();
-                        interceptAbort();
-                        initiated = true;
+        private Init = function () {
+            if (this.supportMonitoring()) {
+                this.instrumentOpen();
+                this.instrumentSetRequestHeader();
+                this.instrumentSend();
+                this.instrumentAbort();
+                this.initiated = true;
```

Comment:
should we catch exception and set initiate to some error state so there will be no more attempts to initiate it?